core/win32/syscall.c

```diff
@@ -1853,8 +1853,12 @@ presys_SetContextThread(dcontext_t *dcontext, reg_t *param_base)
     /* FIXME : we are going to read and write to cxt, which may be unsafe */
     ASSERT(tid != 0xFFFFFFFF);
     LOG(THREAD, LOG_SYSCALLS|LOG_THREADS, IF_DGCDIAG_ELSE(1, 2),
-        "syscall: NtSetContextThread handle="PFX" tid=%d cxt->Xip="PFX"\n",
-        thread_handle, tid, cxt->CXT_XIP);
+        "syscall: NtSetContextThread handle="PFX" tid=%d cxt->Xip="PFX" flags="PFX"\n",
+        thread_handle, tid, cxt->CXT_XIP, cxt->ContextFlags);
+    if (get_thread_id() == tid) {
+        /* Simple case when called on own thread. */
+        return execute_syscall;
```

Comment:
> My case was the simplest one.

Starting small and handling only a subset of context flags is fine, but we have to check for that subset.  The code as submitted directly executes the syscall for any value of flags, with no comment or indication about being incomplete.